lib/cartopy/crs.py

```diff
@@ -1344,6 +1356,59 @@ def quick_vertices_transform(self, vertices, src_crs):
 
         return return_value
 
+    def transform_points(self, src_crs, x, y, z=None):
+        """
+        transform_points(src_crs, x, y[, z])
+
+        Transform the given coordinates, in the given source
+        coordinate system (``src_crs``), to this coordinate system.
+
+        Parameters
+        ----------
+        src_crs
+            instance of :class:`CRS` that represents the
+            coordinate system of ``x``, ``y`` and ``z``.
+        x
+            the x coordinates (array), in ``src_crs`` coordinates,
+            to transform.  May be 1 or 2 dimensional.
+        y
+            the y coordinates (array), in ``src_crs`` coordinates,
+            to transform.  Its shape must match that of x.
+        z: optional
+            the z coordinates (array), in ``src_crs`` coordinates, to
+            transform.  Defaults to None.
+            If supplied, its shape must match that of x.
+
+        Returns
+        -------
+            Array of shape ``x.shape + (3, )`` in this coordinate system.
+
+        """
+        if self == src_crs:
+            # convert from [0,360] to [-180,180]
+            x = (((np.array(x, copy=False) + 180) % 360) - 180)
+        return super().transform_points(src_crs, x, y, z)
+
+    def _project_line_string(self, geometry, src_crs):
```

Comment:
I think that the `transform_points()` addition above is OK, but the project_line_string here is in the wrong place I believe. I think here you are making a line that goes from 178 -> -178, all the way across the map. But, if you put this check into the interpolators/trace.pyx it looks for the boundary and splits the segment into one from 178 -> 180, and another from -180 -> -178. So somehow within trace you'll need to check for PlateCarree to PlateCarree.

Reply:
Moved it to trace.pyx.